src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -285,7 +286,13 @@ protected byte[] engineGetIV() {
   protected AlgorithmParameters engineGetParameters() {
     try {
       AlgorithmParameters parameters = AlgorithmParameters.getInstance("GCM");
-      parameters.init(new GCMParameterSpec(tagLength * 8, iv));
+      byte[] ivForParams = iv;
+      if (ivForParams == null) {
+        // We aren't initialized so we return default and random values
+        ivForParams = new byte[DEFAULT_IV_LENGTH_BYTES];
+        new SecureRandom().nextBytes(ivForParams);
```

Comment:
It's better to use [LibCryptoRng](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/src/com/amazon/corretto/crypto/provider/LibCryptoRng.java#L9) to ensure the secure random provider is backed by AWS-LC.